### PR TITLE
feat(auth): return a proper 401 on auth failure

### PR DESCRIPTION
**Pull request by JeanBarriere**

### CODE REVIEW

File: app/server.js [hunk 1]
```diff
@@ -74,7 +74,10 @@ async function pgSettings(req) {
       [token]
     );
     if (!row) {
-      throw new Error("Invalid or expired token");
+      var error = new Error("InvalidOrExpiredToken");
+      error.statusCode = 401;
+      error.name = "InvalidOrExpiredToken";
```
Comment (judepereira) [inline app/server.js]: Do we need to set error.name since you're already mentioning this in the Error constructor?

Comment (judepereira) [reply app/server.js]: Also, looks like you forgot to throw the error.

Comment (JeanBarriere) [reply app/server.js]: The error parameter is actually the message. GraphQL and PostGraphile are also taking a `statusCode` and a `name` to that object before returning the result. I've put the `error.name` to ensure the request is thrown correctly with the 401

Comment (judepereira) [reply app/server.js]: Ah okay. Cool. Just add this as a comment (since somebody else might be tempted to remove it).

Comment (JeanBarriere) [reply app/server.js]: Well, people always have access to the git history so they know why it's there, but it's done

Comment (wilzbach) [reply app/server.js]: Wait, we already set `.name` with the constructor. What's the purpose of setting it again?

Comment (judepereira) [reply app/server.js]: I can't believe that we had the exact same question @wilzbach!

**Pull request merged**